engine/chain/internal/ef/converter_test.go

```diff
@@ -0,0 +1,140 @@
+		t.Run(name, func(t *testing.T) {
+			tx := txFromBase64(t, test.rawTXBase64)
+
+			converter := ef.NewConverter(test.txGetter)
+			efHex, err := converter.Convert(context.Background(), tx)
+			require.NoError(t, err)
+			require.Equal(t, test.expectedEFHex, efHex)
+
+			// additionally check if converter can convert already-in-ef-hex tx
+			// without the need to fetch source transactions
+			tx, err = sdk.NewTransactionFromHex(efHex)
+			require.NoError(t, err)
+			converter = ef.NewConverter(newMockTransactionsGetter(t, []string{}))
+			efHexRegenerated, err := converter.Convert(context.Background(), tx)
+			require.NoError(t, err)
+			require.Equal(t, efHex, efHexRegenerated)
+		})
+	}
+}
+
+func TestConverterErrorCases(t *testing.T) {
+	tests := map[string]struct {
+		rawTXBase64 string
+		txGetter    *mockTransactionsGetter
+		expectErr   error
+	}{
+		"No source tx provided by TransactionGetter": {
+			rawTXBase64: tx1,
+			txGetter:    newMockTransactionsGetter(t, []string{}).WithOnMissingBehavior(onMissingTxSkip),
+			expectErr:   ef.ErrGetTransactions,
+		},
+		"Not every source tx provided by TransactionGetter": {
+			rawTXBase64: tx2,
+			txGetter: newMockTransactionsGetter(t, []string{
+				tx2Source1,
+			}).WithOnMissingBehavior(onMissingTxSkip),
+			expectErr: ef.ErrGetTransactions,
+		},
+		"TransactionGetter error on missing transaction": {
+			rawTXBase64: tx1,
+			txGetter:    newMockTransactionsGetter(t, []string{}).WithOnMissingBehavior(onMissingTxReturnError),
+			expectErr:   ef.ErrGetTransactions,
+		},
+		"Nil transaction returned by TransactionGetter": {
+			rawTXBase64: tx1,
+			txGetter:    newMockTransactionsGetter(t, []string{}).WithOnMissingBehavior(onMissingTxAddNil),
+			expectErr:   ef.ErrGetTransactions,
+		},
+		"TransactionGetter returned more transactions than requested": {
+			rawTXBase64: tx1,
+			txGetter:    newMockTransactionsGetter(t, []string{tx1Source, tx2Source1}).WithReturnAll(true),
+			expectErr:   ef.ErrGetTransactions,
+		},
+		"TransactionGetter not requested transactions but with correct length": {
+			rawTXBase64: tx1,
+			txGetter:    newMockTransactionsGetter(t, []string{tx2Source1}).WithReturnAll(true),
+			expectErr:   ef.ErrGetTransactions,
+		},
+		"TransactionGetter duplicated transaction": {
+			rawTXBase64: tx2,
+			txGetter:    newMockTransactionsGetter(t, []string{tx2Source1, tx2Source1}).WithReturnAll(true),
+			expectErr:   ef.ErrGetTransactions,
+		},
+	}
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			tx := txFromBase64(t, test.rawTXBase64)
+
+			converter := ef.NewConverter(test.txGetter)
+			efHex, err := converter.Convert(context.Background(), tx)
+			require.ErrorIs(t, err, test.expectErr)
+			require.Empty(t, efHex)
+		})
+	}
+}
+
+func txFromBase64(t *testing.T, rawTXBase64 string) *sdk.Transaction {
+	d, err := base64.StdEncoding.DecodeString(rawTXBase64)
+	require.NoError(t, err)
+	tx, err := sdk.NewTransactionFromBytes(d)
```

Comment:
NewTransactionFromHex wouldn't work here?

Reply:
I've manually converted them from base64 to hex. 
So, as you say, `NewTransactionFromHex` works.

FYI: I have loaded the rawTX using junglenus Json API. And there are base64 encoded strings.